modelscope_agent/tools/code_interpreter/code_interpreter.py

```diff
@@ -15,14 +15,25 @@
 import traceback
 import uuid
 from pathlib import Path
-from typing import Dict, Optional
+from typing import Dict, Literal, Optional, Tuple
 
 import json
 import matplotlib
+import nbformat
 import PIL.Image
 from jupyter_client import BlockingKernelClient
 from modelscope_agent.tools.base import BaseTool, register_tool
 from modelscope_agent.utils.utils import extract_code
+from nbclient import NotebookClient
```

Comment:
what's different between nbclient and jupyter_client? shall we use only one of them, not all of them?

Reply:
nbclient provides a higher-level abstraction for executing Jupyter Notebooks and works with Jupyter Notebook files (.ipynb), and can save the code and whole excuting process in a Jupyter Notebook file (.ipynb) ,  I think it's more more suitable in data science senarios

Reply:
then shall we replace jupyter_client by nbclient at all place?